### ISSUE TITLE
Dataloader not working correctly with Mongoose 6 and 7 

### ISSUE THREAD

**Issue by mareksuscak** [TriPSs/nestjs-query#281]
**Describe the bug**

After the [recent Dataloader fixes](https://github.com/TriPSs/nestjs-query/pull/279), there's now a bug that affects certain versions of Mongoose - namely v6 and v7 because of incompatibility with class-transformer and specifically https://github.com/typestack/class-transformer/issues/991 which causes that values of type `ObjectId` that are not explicitly decorated with `@ObjectId` decorator are randomized during the conversion when their parent entity is being converted to a DTO. This was previously not an issue because `findRelation` and `queryRelations` were reloading the parent entity from the database [here](https://github.com/TriPSs/nestjs-query/blob/v6.1.1/packages/query-mongoose/src/services/reference-query.service.ts?rgh-link-date=2024-07-24T22%3A21%3A31Z#L148) and [here](https://github.com/TriPSs/nestjs-query/blob/v6.1.1/packages/query-mongoose/src/services/reference-query.service.ts?rgh-link-date=2024-07-24T22%3A21%3A31Z#L191) - I'd argue unnecessarily but what's more important is that the entity's fields were always correct despite the previous conversion. But now we only use the DTOs passed into `findRelation` and `queryRelations` and the DTOs all suffer from the class-transformer bug. I'm not sure what's the best way to fix this as it only affects certain versions of Mongoose. We're not seeing this behavior in tests because this repository already uses Mongoose v8 which isn't affected anymore.

### Have you read the [Contributing Guidelines](https://github.com/tripss/nestjs-query/blob/master/CONTRIBUTING.md)?

Yes

**To Reproduce**
See https://github.com/mareksuscak/nestjs-query-bugs/pull/3

**Expected behavior**
ObjectIds do not change randomly and relationships are correctly fetched

**Screenshots**

**Desktop (please complete the following information):**
 - Node Version - v20
 - Nestjs-query Version - v6.1.2

**Additional context**
None

**Comment by TriPSs** [comment]
We can just drop support for the older versions? Don't see why we would need to support those?

**Comment by mareksuscak** [comment]
That's what I was thinking as well @TriPSs. The only question is - do we revert my change in 6.1.3 and then cut 7.0.0, re-introduce the change and drop support for mongoose < 8? The only other alternative I could think of was detecting the installed version of Mongoose and if >5 and <8, we could use the old implementation. It seems to be possible to read the version number by accessing `mongoose.version` - [see the docs](https://mongoosejs.com/docs/check-version.html).

**Comment by TriPSs** [comment]
Nha let's take this moment and drop those versions. I created a PR to revert it.

Could you create a new PR readding it again together with the removal of the older versions?

**Comment by mareksuscak** [comment]
Yes sir 🫡

**Comment by TriPSs** [comment]
Also add a commit with:
```
feat(query-mongoose): Dropped support of older versions

BREAKING CHANGE: Versions 6 and 7 of mongoose are no longer supported
```

This will make sure the bump is correct :)

**Comment by mareksuscak** [comment]
Done @TriPSs https://github.com/TriPSs/nestjs-query/pull/283

**Comment by mareksuscak** [comment]
This was addressed in 6.1.3 (change reverted) and 7.0.0 (change re-introduced and mongoose version range updated).
